### PR TITLE
Fixed GWT Libary Build

### PR DESCRIPTION
**Pull request by cdietrich**

### CODE REVIEW

**Comment by cdietrich** [comment]
please review

**Comment by kthoms** [comment]
LGTM

**Comment by cdietrich** [comment]
@spoenemann @oehme what do you think (from a gradle pov)

File: org.eclipse.xtext.xbase.lib/build.gradle [hunk 1]
```diff
@@ -5,3 +5,23 @@ description = 'The runtime library for Xbase languages.'
 dependencies {
 	compile "com.google.guava:guava:$versions.guava"
 }
+
+task copyGwtSources(dependsOn: 'build') {
+	doLast {
+	
+		ant.copy(toDir: '../org.eclipse.xtext.xbase.lib.gwt/build/gwtSources', verbose: true) {
```
Comment (spoenemann) [inline org.eclipse.xtext.xbase.lib/build.gradle]: Gradle has a Copy task from which you can inherit. I think that's better than using the ant bridge.

Comment (cdietrich) [inline org.eclipse.xtext.xbase.lib/build.gradle]: yes but i did not find any docs or tutorials to do the fitering

Comment (cdietrich) [inline org.eclipse.xtext.xbase.lib/build.gradle]: => i did the copy & delete with ant

Comment (cdietrich) [inline org.eclipse.xtext.xbase.lib/build.gradle]: the original one did a copy with filter only but that does not seem to work with gradle ant builder, at least i did not get it running. => i did the copy & delete both in ant

Comment (cdietrich) [inline org.eclipse.xtext.xbase.lib/build.gradle]: and i did not find a way to do the copy and delte in one task with inheritance

Comment (spoenemann) [inline org.eclipse.xtext.xbase.lib/build.gradle]: [Copy](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.Copy.html) has quite a lot of options. And it's no problem to add a `doLast` block with some custom code (that doesn't overwrite anything from the superclass).

Comment (cdietrich) [inline org.eclipse.xtext.xbase.lib/build.gradle]: yes but none of these seem to do a contains filtering, it all seems be name based filtering and actual content filtering with means of replacements

File: org.eclipse.xtend.lib/build.gradle [hunk 2]
```diff
@@ -6,3 +6,23 @@ dependencies {
 	compile project(':org.eclipse.xtext.xbase.lib')
 	compile project(':org.eclipse.xtend.lib.macro')
 }
+
+task copyGwtSources(dependsOn: 'build') {
```
Comment (oehme) [inline org.eclipse.xtend.lib/build.gradle]: This task should not be here, it belongs in the xtend.lib.gwt project. That project should have a Configuration that includes source jars and unpacks those for inclusion into its own jar. 

Also there is no reason for this task to depend on 'build'. That would mean you run all tests and other checks just to build a source jar.

Comment (cdietrich) [inline org.eclipse.xtend.lib/build.gradle]: @oehme 

yes but i googled that and i did not find any example that does that.

so 

(1) how do i do a source jar config. respectively how do a 

dependencies {
gtwCrap project(':org.eclipse.xtext.xbase.lib', classifier: source)
}

Comment (oehme) [inline org.eclipse.xtend.lib/build.gradle]: This [discussion](https://discuss.gradle.org/t/how-do-i-declare-a-dependency-on-a-modules-test-code/7172/9) solves the same problem for test JARs, you can adjust it to the source JAR problem.

**Comment by cdietrich** [comment]
@oehme can you have a look again. the source jar stuff works now.
is there a easy way to do the contains/not stuff?

**Comment by cdietrich** [comment]
got rid of ant totally

**Comment by oehme** [comment]
LGTM. Duplicate code should be extracted into a script I think.

**Comment by cdietrich** [comment]
did the extraction as well

**Comment by cdietrich** [comment]
@spoenemann @svenefftinge should this go to maintenance?

**Comment by svenefftinge** [comment]
Yes, ideally and then merged into master.
See https://github.com/eclipse/xtext/blob/master/Builds.md#the-release-process

**Pull request closed without merging**